packages/core/src/payment/strategies/stripe-upe/stripe-upe-payment-strategy.ts

```diff
@@ -451,26 +479,34 @@ export default class StripeUPEPaymentStrategy implements PaymentStrategy {
             } = error.body.additional_action_required;
 
             if (type === 'redirect_to_url' && redirect_url) {
-                const { paymentIntent, error: stripeError } =
-                    await this._stripeUPEClient.confirmPayment(
-                        this._mapStripePaymentData(redirect_url),
-                    );
-
-                if (stripeError) {
-                    this._throwDisplayableStripeError(stripeError);
-                    throw new PaymentMethodFailedError();
-                }
+                if (
+                    methodId &&
+                    gatewayId &&
+                    (await this._isPaymentNotComplited(gatewayId, methodId))
+                ) {
```

Comment:
and also a good practice to make your code more simple for reading and try not to use nested logic.
so instead of:
```
if(check1) {
    if (check2) {
        if (check3) {
            execution...
        }
    }
}
```
try to use:
```
if (!check1) {
    return;
}

if (!check2) {
    return;
}

if (!check3) {
    return;
}

execution...

```

Reply:
done